keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1653,8 +1654,48 @@ type QuotaSpec struct {
     //
     // DeviceClassAllowList []string
 }
+
+type MaxDeviceCount struct {
+    // The maximum number of allocated devices. May be zero to prevent using
+    // certain devices.
+    Maximum resource.Quantity
+
+    // Only devices matching all selectors are counted.
+    //
+    // +listType=atomic
+    Selectors []Selector
+}
+
+type MaxQuantity struct {
+    // The maximum sum of a certain quantity attribute. Only allocated devices which
+    // have this attribute as a quantity contribute towards the sum.
+    Maximum resource.Quantity
+
+    // The fully-qualified attribute name ("<domain>/<identifier>").
+    AttributeName string
+}
 ```
 
+<<[UNRESOLVED @pohly]>>
+
+As it stands now, a single ResourceQuota object per namespace can contain all restrictions.
+If we add a new field to the type (like the DeviceLimits or class allow/deny list),
+old schedulers will ignore them and thus not apply the new restrictions.
+
+Should we make QuotaSpec a one-of? Exactly one field would need to be set (true
+or non-empty) per object. This prevents putting all restrictions into a single
+object.
+
+Or shall we nest a one-of inside a slice called "rules" or "settings"?
```

Comment:
Regardless whether we use a slice inside a single ResourceQuota or multiple ResourceQuota objects, do we also want the `MaxDeviceCounts` and `MaxQuantities` slices? We can get repetition also without those.

If we say that a ResourceQuota object contains exactly one restriction, then error messages could refer to only the name of that object and it would be unambiguous which restriction is not satisfied.

Reply:
If we keep the slices, then they probably should have a required name for that purpose.

Reply:
Discussion moved to https://github.com/kubernetes-sigs/wg-device-management/issues/24